dali/base/dautils.cpp

```diff
@@ -47,6 +47,59 @@
 #define MIN_REDIRECTION_LOAD_INTERVAL 1000
 
 
+bool isHostInPlane(IPropertyTree *plane, const char *host, bool ipMatch)
+{
+    Owned<IPropertyTreeIterator> hostsIter;
+    if (plane->hasProp("@hostGroup"))
+    {
+        Owned<IPropertyTree> hostGroup = getHostGroup(plane->queryProp("@hostGroup"), true);
+        hostsIter.setown(hostGroup->getElements("*"));
+    }
+    else
+    {
+        if (!plane->hasProp("hosts"))
+            return false;
+        hostsIter.setown(plane->getElements("hosts"));
+    }
+    SocketEndpoint hostEp;
+    if (ipMatch)
+        hostEp.set(host);
+    ForEach (*hostsIter)
+    {
+        const char *planeHost = hostsIter->query().queryProp(nullptr);
+        if (ipMatch)
+        {
+            SocketEndpoint planeHostEp(planeHost);
+            if (planeHostEp.ipequals(hostEp))
+                return true;
+        }
+        else if (streq(planeHost, host))
+            return true;
+    }
+    return false;
+}
+
+bool getPlaneHost(StringBuffer &host, IPropertyTree *plane, unsigned which)
+{
+    VStringBuffer xpath("hosts[%u]", which+1); // which is 0 based
+    if (plane->hasProp("@hostGroup"))
+    {
+        Owned<IPropertyTree> hostGroup = getHostGroup(plane->queryProp("@hostGroup"), true);
+        if (which >= hostGroup->numChildren())
```

Comment:
should also be checking getCount("hosts").

Reply:
can change it, this is not time critical, but thinking was that numChildren is faster, but I'll change it as more readable if same.